Filters for mobile view

joeytwiddle opened the pull request:
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.

I am just trying to finish off #1793, which fixes #1781.

I will upload a test site in a moment...

joeytwiddle commented:
A generated demo site is here: https://joeytwiddle.github.io/open-event-webapp/MozillaAllHands2017.mobile-view-filters/

Outstanding issues:
- It doesn't work with the dark theme (main dropdown text is white on white)
- Some of the CSS (flexbox) may be too modern for old mobile browsers

agbilotia1998 commented:
@joeytwiddle your travis build is passed, please resolve merge conflicts.

joeytwiddle commented:
Done @agbilotia1998

mariobehling commented:
Thank you!

The dropdown of filters is not visible on Chromium. Please check screenshot below.

![screenshot from 2018-05-23 15-24-14](https://user-images.githubusercontent.com/1583873/40427307-9ac2f38c-5e9d-11e8-9751-695a4ec25ac9.png)

joeytwiddle commented:
I have fixed the issue with the dark theme, and a few other issues.

Test link (same as before): https://joeytwiddle.github.io/open-event-webapp/MozillaAllHands2017.mobile-view-filters/

A new problem I noticed:

- The rooms list appears immediately below its parent button. Because it is wide, it was sticking out the right of the page, causing the page to grow wider.

I tried a "good" solution but didn't get that finished. So I went for a simple solution: Just move the filter button to be the left-most of all the buttons. :wink:

joeytwiddle commented on src/backend/_scss/application.scss:
```diff
@@ -2195,3 +2204,90 @@ a.skip:hover {
+
+.dropdown-inner {
+  max-height: 200px;
+  max-height: 60vh;
```
I am trying to do progressive enhancement here. If the browser does not support "vh" then I think it will just use the previously declared pixel value "200px".

agbilotia1998 commented on src/selenium/schedulePage.js:
```diff
@@ -58,21 +58,27 @@ SchedulePage.toggleMode = function() {
+  }).then(function() {
+    return Promise.all(promiseArr);
   });
-
```
Why is this unit test modified?

joeytwiddle commented:
What has changed today?

- Gave the new dropdown its own classname, so CSS and JS can target it semantically.
- That fixed the click bug above.
- Margins and padding for filter buttons were redefined separately for each different button. I combined them all under the `.filter` class.

I followed a couple of houndci's suggestions, but the remaining suggestions are I think unhelpful. (They would break related styling into separate parts of the file, making related rules harder to find.)

Test link: https://joeytwiddle.github.io/open-event-webapp/MozillaAllHands2017.mobile-view-filters/

Screenshots:

![joey s google-chrome at 9 32 45 pm on sunday 27 may 2018](https://user-images.githubusercontent.com/911799/40586937-629ae4cc-61fb-11e8-87eb-055ea67912fc.png)
![joey s google-chrome at 9 33 33 pm on sunday 27 may 2018](https://user-images.githubusercontent.com/911799/40586938-62e2dc28-61fb-11e8-96b0-148c9e655f18.png)
![joey s google-chrome at 9 46 25 pm on sunday 27 may 2018](https://user-images.githubusercontent.com/911799/40586939-63176de4-61fb-11e8-9629-a2eceabbe640.png)

joeytwiddle commented on src/backend/_scss/application.scss:
```diff
@@ -2200,3 +2194,86 @@ a.skip:hover {
+
+.dropdown-inner {
+  max-height: 200px;
+  max-height: 60vh;
```
This is progressive enhancement

joeytwiddle commented on src/backend/_scss/application.scss:
```diff
@@ -1517,29 +1510,38 @@ a {
 
 .fossasia-filter {
-  height: 33px;
+  margin-top: 4px;
```
This would increase the size of this diff (because a lot of these properties were already out of order)

joeytwiddle commented:
Please merge if there are no outstanding issues. Scroll up for screenshots, test page, and info.

joeysino commented:
Thanks Travis and you @agbilotia1998. So much for last minute tweaks!

joeytwiddle commented:
Thanks Travis and you @agbilotia1998. So much for last minute tweaks!

agbilotia1998 commented:
@joeytwiddle please resolve the conflicts.

joeytwiddle commented:
OK, resolved and rebased.

agbilotia1998 commented:
@joeytwiddle I guess 'Clear Filter' button is not functional in mobile view. Please check it once.
Test Env. - Chrome Browser

joeytwiddle commented:
Hi @agbilotia1998, the only problem I found with the Clear Filters button was that sometimes it didn't appear at all!

That was actually a problem with the filter not being set. If we clicked on one of the room or track list items, but **not** directly on the text, then the menu would be closed without actually doing anything. I fixed that by moving the classes from the title `<span>` up to the wide container `<a>` link.

The new test link is: https://joeytwiddle.github.io/open-event-webapp/MozillaAllHands2017.mobile-view-filters-2/

agbilotia1998 commented:
Screencast:
![mobile-view](https://user-images.githubusercontent.com/21157929/41362311-ad8ce484-6f4e-11e8-998d-6959dbefae80.gif)

joeytwiddle commented:
I have fixed that bug and squashed the commits.

New test URL: https://joeytwiddle.github.io/open-event-webapp/MozillaAllHands2017.mobile-view-filters-5/
